src/ConnectedMode/Migration/IConnectedModeMigration.cs

```diff
@@ -71,4 +62,49 @@ public MigrationProgress(int currentProject, int totalProjects, string message,
         /// </summary>
         public bool IsWarning { get; }
     }
+
+    /// <summary>
+    /// Finds the set of MSBuild files that need to be cleaned as part of the migration
+    /// </summary>
+    internal interface IFileProvider
+    {
+        /// <summary>
+        /// Returns a list of full paths to the set of MSBuild files to be cleaned
+        /// </summary>
+        /// <remarks>The files can be project files or imported props/targets etc</remarks>
+        Task<IEnumerable<string>> GetFilesAsync(CancellationToken token);
```

Comment:
I made a couple of tweaks: I removed the `IProgress` from the file provider and file cleaner.

I think `ConnectedModelMigration` class should be able to raise the higher-level UI progress events, and the cleaner and file provider could just log to the output window.

Reply:
Yea I agree, makes sense that way to me.